### ISSUE TITLE
Add Basemap Layers for Selection

### ISSUE THREAD

**Issue by dloutzenheiser** [MAPC/trailmap3#15]
Currently there is one base map, but need to add and icon to display additional layers (OSM, and imagery).

I do want to select an OSM layer.  Where can I find a catalog of such renderings available in Mapbox?

**Comment by dloutzenheiser** [comment]
This is one good option for OSM
https://www.maptiler.com/maps/#bright//vector/14.17/-71.19405/42.39746

**Comment by theryankelly** [comment]
Interface from XD includes menu and new window with options.  We need to determine what the options will and keep in mind what colors we are choosing for lines.

![Screen Shot 2019-09-06 at 1 58 19 PM](https://user-images.githubusercontent.com/2421382/64453054-aeb0ef80-d0b5-11e9-8147-26af23967ac4.png)
![Screen Shot 2019-09-06 at 1 58 31 PM](https://user-images.githubusercontent.com/2421382/64453055-aeb0ef80-d0b5-11e9-990b-e28a9717fae5.png)

**Comment by atomtay** [comment]
Update: the ability to change the basemap back and forth is all set (still should put together nicer buttons than just "Map 1" and "Map 2" but that shouldn't be too bad). The problem I'm running into is if someone switches base maps after toggling on at least one trail type. On initial load, it looks like the change wipes out those trails, but `fac_type` in the map state doesn't change. Additionally, if you toggle on a different trail, then the previous trails show up as well. The toggled-on trails are there and ready to go, but effectively need to get "re-fired."

I've spent the afternoon on an approach where I take the layer with trails from `prevState`, append it to the list of layers in the new basemap (brought over as `updatedBaseMap` and manipulated as `newMapStyle`), and then set `state.mapStyle` to this new map object:

```
changeBasemap(updatedMapStyle) {
    this.setState((prevState) => {
      const trailLayers = prevState.mapStyle.get('layers').find(layer => layer.get('source') === 'path_overlay');
      let newMapStyle = updatedMapStyle;
      if (trailLayers !== undefined) {
        newMapStyle = updatedMapStyle
          .set('layers', updatedMapStyle.get('layers').push(trailLayers));
      }
      return { mapStyle: newMapStyle };
    });
  }
```

However, I get the following error:
```
:formatted:6432 Error: layers[60]: source "path_overlay" not found
    at Object.pn [as emitValidationErrors] (mapbox-gl.js:29)
    at Re (mapbox-gl.js:33)
    at i.setState (mapbox-gl.js:33)
    at r._updateDiff (mapbox-gl.js:33)
    at r._diffStyle (mapbox-gl.js:33)
    at r.setStyle (mapbox-gl.js:33)
    at StaticMap._updateMapStyle (static-map.js:165)
    at StaticMap.componentDidUpdate (static-map.js:133)
    at commitLifeCycles (react-dom.development.js:20064)
    at commitLayoutEffects (react-dom.development.js:22813) "
    in StaticMap (created by InteractiveMap)
    in InteractiveMap (at map.jsx:94)
    in Map (created by Router.Consumer)
    in Router.Consumer (created by Route)
    in Route (at App.jsx:10)
    in App (created by Router.Consumer)
    in Router.Consumer (created by Route)
    in Route (at pages/​index.jsx:9)"
```
(Here, "path_overlay" is the layer source I'm trying to bring over. Another one of note is "proposed_overlay" for proposed trails, but I'm just working on set paths right now and will abstract later).

Wondering if I need to add "path_overlay" as an acceptable source to `newMapStyle` before adding the layer? Definitely need to dive more into the MapboxGL documentation to figure this one out.

**Comment by atomtay** [comment]
Resolved in [commit 479b62d](https://github.com/MAPC/trailmap3/commit/479b6d25067d308e2f686ed3d01ee6da1c2898b6)
